openshift/topics/advanced_concepts.adoc

```diff
@@ -282,12 +282,76 @@ route "sso" patched
 ----
 
 [[sso-connecting-to-an-external-database]]
-==== Connecting to an external database
+=== Connecting to an external database
 
 {project_name} can be configured to connect to an external (to OpenShift cluster) database. In order to achieve this, you need to modify the `sso-{database name}` Endpoints object to point to the proper address. The procedure is described in the https://docs.openshift.com/container-platform/latest/networking/configuring-ingress-cluster-traffic/configuring-ingress-cluster-traffic-service-external-ip.html#nw-service-external-ip_configuring-ingress-cluster-traffic-service-external-ip[OpenShift manual].
 
 Tip: The easiest way to get started is to deploy {project_name} from a template and then modify the Endpoints object. You might also need to update some of the datasource configuration variables in the DeploymentConfig. Once you're done, just roll a new deployment out.
 
+[[sso-using-custom-jdbc-driver]]
+=== Using Custom JDBC Driver
+
+To connect to any database, the JDBC driver for that database must be present and {project_name} configured
+properly. Currently, the only JDBC driver available in the image is the PostgreSQL JDBC driver. For any other
+database, you need to extend the {project_name} image with a custom JDBC driver and a CLI script to register
+it and set up the connection properties. The following steps illustrate how to do that, taking MariaDB driver
+as an example. Update the example for other database drivers accordingly.
+
+. Create an empty directory.
+. Download the JDBC driver binaries into this directory.
+. Create a new `Dockerfile` file in this directory with the following contents. For other databases, replace `mariadb-java-client-2.5.4.jar` with the filename of the respective driver:
++
+[source,subs="attributes+,macros+,+quotes"]
+----
+*FROM* {openshift_image}:latest
+
+*COPY* sso-extensions.cli /opt/eap/extensions/
+*COPY* mariadb-java-client-2.5.4.jar /opt/eap/extensions/jdbc-driver.jar
+----
+. Create a new `sso-extensions.cli` file in this directory with the following contents. Update the values of the variables in italics according to the deployment needs:
++
+[source,bash,subs="attributes+,macros+,+quotes"]
+----
+batch
+
+set DB_DRIVER_NAME=_mariadb_
+set DB_USER=_username_
```

Comment:
@hmlnarik I would probably use `DB_USERNAME` instead of `DB_USER` (just for consistency what's currently in the templates / what's expected / recognized as environment variable by the image)